### ISSUE TITLE
Kubernetes / OpenShift integration certificate verify failed

### ISSUE THREAD

**Issue by edbergavera** [sameersbn/docker-gitlab#1125]
Hello,

We are having issues activating kubernets/openshift integration in GitLab 8.16.6 with following error below: 

`We tried to send a request to the provided URL but an error occurred: HTTP status code , SSL_connect returned=1 errno=0 state=SSLv3 read server certificate B: certificate verify failed`

The SSL certificate on the kubernetes api endpoint is properly setup with CA bundle certificate.

**Comment by paolomainardi** [comment]
Are you sure to use the right certificate ? I am currently using the K8s integration with gke without any kind of problem.

**Comment by andrewmyhre** [comment]
I'm having the same problem. We're using a self-signed certificate for our kubernetes cluster.
In the 'Custom CA Bundle' field in Gitlab I've tried providing the ca.crt contents from our 'gitlab' service account's secret, and I've also tried using the CA for the kubernetes api itself. Both come back with 'Certificate verify failed'. We're using Gitlab 9.0 so the message is slightly different:

```
We tried to send a request to the provided URL but an error occurred: HTTP status code , SSL_connect returned=1 errno=0 state=error: certificate verify failed
```

I'm wondering if it's to do with the fact that our certificate is self-signed and doesn't have a valid chain, so it won't pass verification.
